icu4j/main/classes/core/src/com/ibm/icu/util/ULocale.java

```diff
@@ -585,7 +593,17 @@ public static ULocale getDefault() {
             if (!defaultLocale.equals(currentDefault)) {
                 defaultLocale = currentDefault;
                 defaultULocale = forLocale(currentDefault);
-            }
+
+                if (!JDKLocaleHelper.hasLocaleCategories()) {
+                    // Detected Java default Locale change.
+                    // We need to update category defaults to match
+                    // Java 7's behavior on Android API level 21..23.
+                    for (Category cat : Category.values()) {
+                        int idx = cat.ordinal();
+                        defaultCategoryLocales[idx] = currentDefault;
+                        defaultCategoryULocales[idx] = forLocale(currentDefault);
+                    }
+                }            }
```

Comment:
Nit: Two lines have been joined into one here.

Reply:
Sorry, bad cleanup after copy & paste, but I will just leave this one as is for now.